### PR TITLE
Add customer verification CLI commands and refactor verification logic

### PR DESCRIPTION
**Pull request by delano**
## Summary

Introduces a unified operation for managing customer verification state across Redis and SQL stores, with new CLI commands to verify/unverify customers manually.

### Changes

**New Operation: `Auth::Operations::SetCustomerVerification`**
- Centralizes customer verification state changes for both Redis (Familia) and SQL (Rodauth accounts)
- Supports both "simple" auth mode (Redis only) and "full" auth mode (SQL + Redis)
- Includes `rodauth_already_synced` parameter to skip redundant SQL updates when called from Rodauth hooks
- Implements idempotency: returns `:no_change` when already in target state
- Proper error handling with custom exceptions (`NoAuthDatabase`, `AccountNotFound`)
- Comprehensive logging of state changes

**New CLI Commands**
- `bin/ots customers verify EMAIL|EXTID` — Mark customer as verified
- `bin/ots customers unverify EMAIL|EXTID` — Mark customer as unverified
- Both commands support lookup by email or extid
- Friendly error messages for missing customers, anonymous accounts, and auth DB issues
- Exit codes: 0 for success/no-change, 1 for errors

**Refactored Rodauth Hook**
- Replaced inline `VerifyCustomer` operation with new `SetCustomerVerification`
- Passes `rodauth_already_synced: true` to skip redundant SQL write (Rodauth already committed status_id=2)
- Maintains same behavior: verifies customer via 'email' provenance tag

**Removed**
- `Auth::Operations::VerifyCustomer` — superseded by more general `SetCustomerVerification`

### Design Notes

Cross-store consistency is achieved through application-level ordering: SQL update first (inside transaction), Redis save second. This ensures clean rollback on SQL failure while keeping the auth state (SQL) as the source of truth.

## Test Plan

Added comprehensive unit tests:
- `apps/web/auth/spec/operations/set_customer_verification_spec.rb` — 288 lines covering idempotency, simple/full auth modes, error cases, and the `rodauth_already_synced` contract
- `spec/cli/customers_command_spec.rb` — CLI integration tests for verify/unverify commands, error handling, and edge cases

All tests pass with the new implementation.

https://claude.ai/code/session_017nfy42FfzgYavVqYwhpx1f

### CODE REVIEW

**Comment by delano** [comment]
## Manual test plan

Three flows to exercise. The third is the one that previously went through `VerifyCustomer` — it now goes through `SetCustomerVerification` with `rodauth_already_synced: true`, so it deserves explicit attention to confirm behavior is unchanged.

### 1. New CLI flow — simple mode

```bash
# Drop into a clean test customer
bin/ots customers create cli-simple@example.com --no-verified
bin/ots customers show cli-simple@example.com           # verified: false

bin/ots customers verify cli-simple@example.com         # "Verified: c***@example.com"
bin/ots customers show cli-simple@example.com           # verified: true, verified_by: cli_provision

bin/ots customers verify cli-simple@example.com         # "already verified"; exit 0
echo $?                                                 # 0

bin/ots customers unverify cli-simple@example.com
bin/ots customers show cli-simple@example.com           # verified: false, verified_by: nil

# Error paths
bin/ots customers verify missing@example.com            # "Customer not found"; exit 1
EXTID=$(bin/ots customers show cli-simple@example.com | grep -i extid | awk '{print $2}')
bin/ots customers verify "$EXTID"                       # lookup-by-extid works
```

### 2. New CLI flow — full auth mode

```bash
# Same as simple, plus the SQL accounts row should change too
AUTHENTICATION_MODE=full bin/ots customers create cli-full@example.com --no-verified
AUTHENTICATION_MODE=full bin/ots customers verify cli-full@example.com

psql "$AUTH_DATABASE_URL" -c \
  "SELECT email, status_id FROM accounts WHERE email='cli-full@example.com';"
# status_id = 2

AUTHENTICATION_MODE=full bin/ots customers unverify cli-full@example.com
psql "$AUTH_DATABASE_URL" -c \
  "SELECT email, status_id FROM accounts WHERE email='cli-full@example.com';"
# status_id = 1

# Loud-failure path: customer in Redis with no matching Rodauth row
bin/ots customers create redis-only@example.com --no-verified
AUTHENTICATION_MODE=full bin/ots customers verify redis-only@example.com
# Error: No Rodauth account for redis-only@example.com.
# Run `bin/ots customers sync-auth-accounts` to reconcile. exit 1
```

### 3. Existing email-verification flow (previously `VerifyCustomer`)

This is the path the refactor changes mechanically — Rodauth's `after_verify_account` hook now resolves the customer inline inside its existing `safe_execute` block and calls `SetCustomerVerification.new(..., rodauth_already_synced: true)` instead of `VerifyCustomer.new(account:)`. The contract is: skip our own SQL update (Rodauth already committed `status_id=2`), only mirror to Redis.

Steps:

```bash
# Full auth mode so the Rodauth verification flow actually runs
export AUTHENTICATION_MODE=full

# 1. Sign up via the web UI (POST /create-account) with email + password.
#    Or curl-equivalent against your local Rodauth endpoint.

# 2. Capture the verification key (avoiding the email round-trip)
psql "$AUTH_DATABASE_URL" -c \
  "SELECT a.email, vk.key FROM accounts a
   JOIN account_verification_keys vk ON vk.id = a.id
   WHERE a.email = 'hook-test@example.com';"

# 3. Hit the verification URL (browser, or POST /verify-account with the key).
#    Rodauth flips status_id=2 in its own transaction, then fires
#    after_verify_account.

# 4. Confirm Customer side mirrored:
bin/ots customers show hook-test@example.com
# Expect: verified: true, verified_by: email

# 5. Confirm the SQL accounts row was changed exactly once (by Rodauth,
#    not by our op). updated_at should reflect a single update around the
#    verification time:
psql "$AUTH_DATABASE_URL" -c \
  "SELECT email, status_id, updated_at FROM accounts
   WHERE email='hook-test@example.com';"
# status_id = 2 (Rodauth's update; no double-tap from us)

# 6. Confirm the new op log line shows the contract parameter:
grep set-customer-verification log/auth.log | tail -1
# Expected shape:
#   [set-customer-verification] cust_... verified=true verified_by="email"
#   auth_mode=full rodauth_already_synced=true
```

**What to look for as a regression of the refactor:**
- `verified_by` should be `'email'` (not `'cli_provision'` or anything else).
- The log line must show `rodauth_already_synced=true` for hook-triggered verifications. If it shows `false`, we'd be issuing a redundant `UPDATE accounts` inside Rodauth's transaction.
- A failed Customer save in this path should still be caught by `safe_execute` and not bubble up to the user — verification should complete from the user's perspective even if the Redis mirror fails. To exercise this, you can stub `Onetime::Customer#save` to raise and confirm the user sees a successful verification page while the error is logged.

### Quick spec runs

```bash
pnpm run test:rspec apps/web/auth/spec/operations/set_customer_verification_spec.rb
pnpm run test:rspec spec/cli/customers_command_spec.rb
```

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/onetime/cli/customers/unverify_command.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+
+      argument :identifier,
+        type: :string,
+        required: true,
+        desc: 'Email address or extid of the customer'
+
+      def call(identifier:, **)
+        boot_application!
+
+        normalized = OT::Utils.normalize_email(identifier)
```
[the remark this answers is not in the document]
Comment (delano) [reply lib/onetime/cli/customers/unverify_command.rb]: Investigated and not making this change. Extids in this codebase are pure lowercase ASCII — the `Customer` model declares `feature :external_identifier, format: 'ur%<id>s'` (`lib/onetime/models/customer.rb:67`), so they look like `ur<lowercase-hex>`. `OT::Utils.normalize_email` is `email.to_s.strip.unicode_normalize(:nfc).downcase(:fold)` (`lib/onetime/utils/strings.rb:118-120`), which is a no-op on already-stripped, already-lowercase, pure-ASCII input.

Also matches the established pattern at `lib/onetime/cli/customers/show_command.rb:39`, which applies `normalize_email` unconditionally before `Customer.load_by_extid_or_email` and has been in production. Adding a conditional branch here would diverge from that convention without addressing a real bug.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: lib/onetime/cli/customers/verify_command.rb [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+      argument :identifier,
+        type: :string,
+        required: true,
+        desc: 'Email address or extid of the customer'
+
+      def call(identifier:, **)
+        boot_application!
+
+        normalized = OT::Utils.normalize_email(identifier)
```
[the remark this answers is not in the document]
Comment (delano) [reply lib/onetime/cli/customers/verify_command.rb]: Same response as the `unverify_command.rb` thread: not making this change. Extids in this codebase are pure lowercase ASCII (`ur<lowercase-hex>` per the `Customer` model's `feature :external_identifier, format: 'ur%<id>s'` declaration at `lib/onetime/models/customer.rb:67`), so `normalize_email` is a no-op on them. Matches the existing `show_command.rb:39` pattern, which applies it unconditionally before the same `load_by_extid_or_email` lookup.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/web/auth/operations/set_customer_verification.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+      #   (not raised when rodauth_already_synced: true)
+      # @raise [AccountNotFound] full auth mode + no accounts row for email
+      #   (not raised when rodauth_already_synced: true)
+      def call
+        return :no_change if @customer.verified? == @verified
+
+        update_rodauth_account! if full_auth_mode? && !@rodauth_already_synced
+        update_customer!
+
+        auth_logger.info "[set-customer-verification] #{@customer.objid} " \
```
[the remark this answers is not in the document]
Comment (delano) [reply apps/web/auth/operations/set_customer_verification.rb]: Switched to `@customer.extid` in `47050f0`, though for a different reason than stated. The codebase doesn't actually have a "don't log internal IDs" policy — `apps/web/core/logic/authentication/authenticate_session.rb:30,49,58,89,111` routinely logs `customer.objid` as `user_id`/`customer_id`.

The real reason `extid` is the better choice here: this op straddles two stores, and `extid` is the cross-store correlation key (it's what `accounts.external_id` stores on the Rodauth side). For an auth-domain log line where someone is debugging a verification-state divergence, `extid` is the identifier they'll already have in hand.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: apps/web/auth/operations/set_customer_verification.rb [hunk 4]
```diff
@@ -0,0 +1,115 @@
...
+      end
+
+      def update_rodauth_account!
+        db = @db || Auth::Database.connection
+        raise NoAuthDatabase, 'Auth database unreachable' unless db
+
+        rows = db.transaction do
+          db[:accounts]
+            .where(email: @customer.email)
+            .update(status_id: @verified ? 2 : 1, updated_at: Time.now)
```
[the remark this answers is not in the document]
Comment (delano) [reply apps/web/auth/operations/set_customer_verification.rb]: Partial accept in `47050f0`:

**Timestamp** — went one step further than `Time.now.utc` and switched to `Sequel::CURRENT_TIMESTAMP`. This lets the database own `updated_at` directly, removing client-side time generation from the equation entirely (no TZ to drift). Matches the prior art at `lib/onetime/cli/customers/sync_auth_accounts_command.rb:260,280`.

**Magic numbers** — not introducing constants. The `1`/`2` literals are pervasive across the codebase (`lib/onetime/cli/customers/create_command.rb:154`, all of `apps/web/auth/spec/integration/migrations_*_spec.rb`, etc.); the only `STATUS_VERIFIED`/`STATUS_UNVERIFIED` constants are local helpers inside `apps/web/auth/spec/spec_helper.rb` for test-DB setup. Introducing a constant in this one operation file would be locally inconsistent and wouldn't deliver the maintainability win without a broader cleanup. Instead, I added an inline comment documenting the mapping, matching the documentation pattern at `create_command.rb:154`. Happy to follow up on a codebase-wide refactor as a separate issue if the team wants it.

---
_Generated by [Claude Code](https://claude.ai/code)_
